packages/o-spreadsheet-engine/src/helpers/figures/charts/runtime/chart_highlight.ts

```diff
@@ -0,0 +1,123 @@
+import { ChartDataset } from "chart.js";
+import { Color } from "../../../..";
+import { ChartType } from "../../../../types/chart";
+import { setColorAlpha } from "../../../color";
+
+export function getHighlightsHelpers(type: ChartType) {
+  switch (type) {
+    case "scatter":
+    case "radar":
+    case "line":
+      return {
+        highlightItem: highlightLineChartItem,
+        unHighlightItems: unHighlightLineChartItems,
+      };
+    case "bar":
+      return {
+        highlightItem: highlightBarChartItem,
+        unHighlightItems: unHighlightBarChartItems,
+      };
+    case "pie":
+      return {
+        highlightItem: highlightPieChartItem,
+        unHighlightItems: unHighlightPieChartItems,
+      };
+    case "combo":
+      return {
+        highlightItem: highlightComboChartItem,
+        unHighlightItems: unHighlightComboChartItems,
+      };
+    default:
+      return {
+        highlightItem: () => {},
+        unHighlightItems: () => {},
+      };
+  }
+}
+
+function highlightPieChartItem(item: { index?: number }, dataSets: ChartDataset[]) {
+  const index = item.index;
+  dataSets.forEach((dataset) => {
```

Comment:
```suggestion
  for(const dataset of datasets)```

some people don't like foreach in here